backend/assignments/2/src/main/java/com/kdu/caching/controller/GeocodingController.java

```diff
@@ -0,0 +1,47 @@
+package com.kdu.caching.controller;
+
+import com.kdu.caching.dto.forwardgeocoding.ForwardGeocodingRequest;
+import com.kdu.caching.exceptions.ReverseGeocodingFailureException;
+import com.kdu.caching.model.GeocodingData;
+import com.kdu.caching.model.ReverseGeocodingData;
+import com.kdu.caching.service.GeocodingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class GeocodingController {
+
+    private final GeocodingService geocodingService;
+
+    @Autowired
+    public GeocodingController(GeocodingService geocodingService) {
+        this.geocodingService = geocodingService;
+    }
+
+    @GetMapping("/geocoding")
+    public ResponseEntity<Object> forwardGeocoding(@RequestParam String address) {
```

Comment:
If your response object is known then pass the specifics like `ResponseEntity<GeocodingData>`